File: src/main/java/net/lomeli/lomlib/client/addon/jei/crafting/ShapelessFluidRecipeWrapper.java
package net.lomeli.lomlib.client.addon.jei.crafting;

import mezz.jei.api.recipe.wrapper.ICraftingRecipeWrapper;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import java.util.Collections;
import java.util.List;

import net.minecraft.client.Minecraft;
import net.minecraft.item.ItemStack;

import net.lomeli.lomlib.client.addon.jei.BlankRecipeWrapper;
import net.lomeli.lomlib.core.recipes.ShapelessFluidRecipe;

public class ShapelessFluidRecipeWrapper extends BlankRecipeWrapper implements ICraftingRecipeWrapper {
    @Nonnull
    private final ShapelessFluidRecipe recipe;

    public ShapelessFluidRecipeWrapper(ShapelessFluidRecipe recipe) {
        this.recipe = recipe;
        for (Object input : this.recipe.getInput()) {
            if (input instanceof ItemStack) {
                ItemStack itemStack = (ItemStack) input;
                if (itemStack.stackSize > 1) {
                    itemStack.stackSize = 1;
                }
            }
        }
    }

    @Nonnull
    @Override
    public List getInputs() {
        return recipe.getInput();
    }

    @Nonnull
    @Override
    public List<ItemStack> getOutputs() {
        return Collections.singletonList(recipe.getRecipeOutput());
    }

    @Override
    public void drawAnimations(@Nonnull Minecraft minecraft, int recipeWidth, int recipeHeight) {

    }

    @Nullable
    @Override
    public List<String> getTooltipStrings(int mouseX, int mouseY) {
        return null;
    }

    @Override
    public void drawInfo(@Nonnull Minecraft minecraft, int recipeWidth, int recipeHeight, int mouseX, int mouseY) {

    }

    @Override
    public boolean handleClick(@Nonnull Minecraft minecraft, int mouseX, int mouseY, int mouseButton) {
        return false;
    }
}

File: src/main/kotlin/net/lomeli/lomlib/client/addon/jei/crafting/ShapelessFluidRecipeWrapper.kt
package net.lomeli.lomlib.client.addon.jei.crafting

import mezz.jei.api.ingredients.IIngredients
import mezz.jei.api.recipe.BlankRecipeWrapper
import mezz.jei.api.recipe.wrapper.ICraftingRecipeWrapper
import java.util.Collections

import net.minecraft.client.Minecraft
import net.minecraft.item.ItemStack

import net.lomeli.lomlib.core.recipes.ShapelessFluidRecipe

class ShapelessFluidRecipeWrapper(private val recipe: ShapelessFluidRecipe) : BlankRecipeWrapper(), ICraftingRecipeWrapper {

    init {
        for (input in this.recipe.getInput()) {
            if (input is ItemStack && input.stackSize > 1)
                input.stackSize = 1
        }
    }

    override fun getInputs(): List<*> {
        return recipe.getInput()
    }

    override fun getOutputs(): List<ItemStack> {
        return listOf(recipe.recipeOutput!!)
    }

    override fun drawAnimations(minecraft: Minecraft, recipeWidth: Int, recipeHeight: Int) {
    }

    override fun getTooltipStrings(mouseX: Int, mouseY: Int): List<String>? {
        return null
    }

    override fun drawInfo(minecraft: Minecraft, recipeWidth: Int, recipeHeight: Int, mouseX: Int, mouseY: Int) {
    }

    override fun handleClick(minecraft: Minecraft, mouseX: Int, mouseY: Int, mouseButton: Int): Boolean {
        return false
    }

    override fun getIngredients(ingredients : IIngredients?) {
    }
}
